libkpod/image/copy.go

```diff
@@ -152,9 +158,20 @@ func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *typ
 				}
 			}
 		}
-	} else if splitArr[0] == "oci" {
-		// needs to be implemented in future
-		return errors.Errorf("oci not supported")
+	} else if splitArr[0] == "oci-archive" {
```

Comment:
Should these be constants rather then strings?  Are we using oci or oci-archive?

Reply:
oci-archive right? Since the new transport is called oci-archive

Reply:
Are these defined as constants anywhere?  We should check for oci-archive or docker-archive and then fail otherwise

Reply:
This code is in `PullImage`, which is also used for kpod pull. So if docker-archive, and oci-archive are not given, then it should just do a normal pull from a registry.

Reply:
I’d prefer if this didn’t do string parsing of `imgName`:

1. It is incorrect in the `defaultName := DefaultRegistry + imgName` case; the first colon would separate hostname and port, or host/repo and tag, not transport and transport-specific data
2. It is unnecessarily embedding the syntax and parsing logic: after `srcRef` has been successfully set, use `srcRef.Transport()`.  Then the code could either compare them directly (`srcRef.Transport() == ociarchive.Transport`), or perhaps use name comparisons against the c/image-provided data (`srcRef.Transport().Name() == ociarchive.Transport.Name()`).